dm/templates/gke/gke.py

```diff
@@ -116,15 +117,17 @@ def generate_config(context):
         'servicesIpv4Cidr'
     ]
 
-    if (
-        # https://github.com/GoogleCloudPlatform/deploymentmanager-samples/issues/463
-        propc.get('enableDefaultAuthOutput', False) and (
-            version.parse(propc.get('initialClusterVersion').split('-')[0]) < version.parse("1.12") or
-            propc.get('masterAuth', {}).get('clientCertificateConfig', False)
-        )
-    ):
-        output_props.append('clientCertificate')
-        output_props.append('clientKey')
+    initial_cluster_version = propc.get('initialClusterVersion')
+    if initial_cluster_version.lower() != 'latest':
+        if (
+            # https://github.com/GoogleCloudPlatform/deploymentmanager-samples/issues/463
+            propc.get('enableDefaultAuthOutput', False) and (
```

Comment:
This doesn't look quite right.

According to the existing logic, if I have "latest" (ie >= 0.12) *and* propc.get('masterAuth', {}).get('clientCertificateConfig', False), then we should still append these output props.

Could you update this to make the logic on this clearer. Instead of putting everything in a large block, I'd suggest creating a greater_than_012 conditional (which is also true on latest).

Reply:
You are right, done.